socorro/external/es/index_cleaner.py

```diff
@@ -68,13 +74,23 @@ def delete_old_indices(self):
                 # This index doesn't look like a crash index, let's skip it.
                 continue
 
-            # This won't take the week part of our indices into account...
-            index_date = datetime.datetime.strptime(
-                index,
-                self.config.elasticsearch.elasticsearch_index
-            )
-            # So we need to get that differently, and then add it to the date.
-            index_date += datetime.timedelta(weeks=int(index[-2:]))
+            if predicate is None or predicate(index):
+                index_client.delete(index)
+
+    def delete_old_indices(self):
+        self.delete_indices(self.is_index_old)
+
+    def is_index_old(self, index):
+        now = utc_now()
+        policy_delay = datetime.timedelta(weeks=self.config.retention_policy)
+        time_limit = (now - policy_delay).replace(tzinfo=None)
+
+        # This won't take the week part of our indices into account...
+        index_date = datetime.datetime.strptime(
+            index,
+            self.config.elasticsearch.elasticsearch_index
+        )
+        # So we need to get that differently, and then add it to the date.
+        index_date += datetime.timedelta(weeks=int(index[-2:]))
```

Comment:
It took me a while to figure out what this is about until I tried it all by hand. Ugh. I wonder why it doesn't do the week part.

The last part of this line assumes that the last two characters are the week, but that's configured by `elasticsearch_index`. I'm pretty sure we use some variation of the default everywhere, but I'm not positive. I'm not sure how to do it better. I think I'd give up and add a note along these lines:

```
NOTE(osmose): This assumes that the last two characters of the index are the week.
```

or something like that.

Reply:
Stackoverflow pointed me to the [docs](https://docs.python.org/2/library/datetime.html):

> When used with the strptime() method, %U and %W are only used in calculations when the day of the week and the year are specified.

Reply:
Ahh... ok. That makes sense.